### PR TITLE
added deleteTag

### PR DESCRIPTION
**Pull request by xuhas**
Hi, this is my first try at contributing and I tried to tackle the feature request for deleting tags in a given git repo ([#74147](https://github.com/microsoft/vscode/issues/74147)) .
My changes are based on the createTag command.

Let me know if there is something I could change to make this better.

### CODE REVIEW

**Review by Drockk — APPROVED** [review]
For me, everything is ok

File: extensions/git/src/commands.ts [hunk 1]
```diff
@@ -1631,6 +1631,22 @@ export class CommandCenter {
 		await repository.tag(name, message);
 	}
 
+	@command('git.deleteTag', { repository: true })
+	async deleteTag(repository: Repository): Promise<void> {
+		const inputTagName = await window.showInputBox({
```
Comment (joaomoreno) [inline extensions/git/src/commands.ts]: We shouldn't be asking the user to provide a tag name here. We should show the user a picker to choose which tag should be deleted.

**Review by joaomoreno — CHANGES_REQUESTED** [review]

File: extensions/git/src/commands.ts (hunk 1 above)
Comment (xuhas) [reply extensions/git/src/commands.ts]: Good point, I will try to implement the suggested solution in the coming week. Thank you for the feedback.

File: extensions/git/src/commands.ts [hunk 2]
```diff
@@ -38,6 +38,25 @@ class CheckoutItem implements QuickPickItem {
 	}
 }
 
+class TagItem implements QuickPickItem {
+
+	get label(): string { return (this.tag.name || '').substr(0, 20); }
```
Comment (xuhas) [inline extensions/git/src/commands.ts]: Hi @joaomoreno , I would like your input on the label length allowed for the quick pick items. For now, I am arbitrarily cropping the tag label to 20 characters , this rest of the line is taken by the tag message. I am wondering if 20 characters is ok?

**Comment by xuhas** [comment]
Hi @joaomoreno , I made the requested changes by adding a quick pick Item. Let me know if there is anything I could do to make this better.

File: extensions/git/src/commands.ts (hunk 2 above)
Comment (joaomoreno) [reply extensions/git/src/commands.ts]: Sure.

File: extensions/git/package.json [hunk 3]
```diff
@@ -267,6 +267,16 @@
         "title": "%command.createTag%",
         "category": "Git"
       },
+      {
+        "command": "git.deleteTag",
+        "title": "%command.deleteTag%",
+        "category": "Git"
+      },
+      {
+        "command": "git.getTags",
```
Comment (joaomoreno) [inline extensions/git/package.json]: This is not needed.

File: extensions/git/package.json [hunk 4]
```diff
@@ -555,6 +565,14 @@
           "command": "git.createTag",
           "when": "config.git.enabled && gitOpenRepositoryCount != 0"
         },
+        {
+          "command": "git.deleteTag",
+          "when": "config.git.enabled && gitOpenRepositoryCount != 0"
+        },
+        {
+          "command": "git.getTags",
```
Comment (joaomoreno) [inline extensions/git/package.json]: Nor is this.

File: extensions/git/src/git.ts [hunk 5]
```diff
@@ -1285,6 +1285,26 @@ export class Repository {
 		await this.run(args);
 	}
 
+	async deleteTag(name: string): Promise<void> {
+		let args = ['tag', '-d', name];
+		await this.run(args);
+	}
+
+	async getTags(): Promise<Tag[]> {
```
Comment (joaomoreno) [inline extensions/git/src/git.ts]: This is not needed. We already have a list of tags in `repository.refs`. I see that this brings in the tag message, but let's assume the user's not really interested in that at the point that they wish to delete a tag.

**Review by joaomoreno — CHANGES_REQUESTED** [review]

**Comment by joaomoreno** [comment]
I've cleaned it up and made it simpler. Great job @xuhas, thanks a lot! :beers:

**Pull request merged**